src/components/discovery/VariantSearchHeader.js

```diff
@@ -6,40 +6,92 @@ import { useSelector } from "react-redux";
+        ? Object.keys(variantsOverviewResults?.assemblyIDs)
+        : [];
 
-  // or default to GRCh37?
-    const [lookupAssemblyId, setLookupAssemblyId] = useState(null);
     const assemblySchema = dataType.schema?.properties?.assembly_id;
     const genotypeSchema = dataType.schema?.properties?.calls?.items?.properties?.genotype_type;
 
   // hardcoded style from DiscoverySearchForm, change to params
     const labelCol = {lg: { span: 24 }, xl: { span: 4 }, xxl: { span: 3 }};
     const wrapperCol = {lg: { span: 24 }, xl: { span: 20 }, xxl: { span: 18 }};
 
-    // obtain ref and alt values from state
-    const form = useSelector(state =>
-        state.explorer?.dataTypeFormsByDatasetID[Object.keys(state.explorer.dataTypeFormsByDatasetID)[0]]);
-    const activeRefValue = form[0].formValues?.conditions === undefined ? "" : form[0].formValues.conditions.filter(c =>
-        c.value.field === "[dataset item].reference"
-    )[0].value.searchValue;
-    const activeAltValue = form[0].formValues?.conditions === undefined ? "" : form[0].formValues.conditions.filter(c =>
-        c.value.field === "[dataset item].alternative"
-    )[0].value.searchValue;
-    const handleAssemblyIdChange = (value) => {
+    const helpText = {
+        "assemblyId": assemblySchema?.description,
+        "genotype": genotypeSchema?.description,
+        "locus": "Enter gene name (eg \"BRCA1\") or position (\"chr17:41195311-41278381\")",
+        "ref/alt": "Combination of nucleotides A, C, T, and G, including N as a wildcard - i.e. AATG, CG, TNN"
+    };
+
+    // custom validation since this form isn't submitted, it's just used to fill fields in hidden form
+    // each field is validated invididually elsewhere
+    // for final validation, we only need to make sure required fields are non-empty
+    const validateVariantSearchForm = () => {
+
+      // check assembly
+        if (!assemblyId) {
+        // change assemblyId helptext & outline
+            setFieldsValidity({...fieldsValidity, "assemblyId": false});
+        }
+
+        // check locus
+        const {chrom, start, end} = locus;
+        if (!chrom || !start || !end) {
+        // change locus helptext & outline
+            setFieldsValidity({...fieldsValidity, "locus": false});
+        }
+    };
+
+    useEffect(() => {
+        if (isSubmitting) {
+            validateVariantSearchForm();
+        }
+    }, [isSubmitting]);
+
 
+    const isValidLocus = (locus) => {
+        return locus.chrom !== null  && locus.start !== null && locus.end !== null;
+    };
+
+    const setLocusValidity = (isValid) => {
+        setFieldsValidity({...fieldsValidity, "locus": isValid});
+    };
+
+    const handleLocusChange = (locus) => {
+        const {chrom, start, end} = locus;
+
+        if (isValidLocus(locus)) {
+            setLocusValidity(true);
+
+        } else {
+            setLocusValidity(false);
+            return;
+        }
+
+        setLocus({chrom: chrom, start: start, end: end});
```

Comment:
this pattern of constructing a reduced form of the locus (locus, start, end) seems to pop up a few times - could be worth making a function to build it.

also, this again could be written more compactly as `{chrom, start, end}`